common/meta/raft-store/src/state_machine/sm.rs

```diff
@@ -451,6 +453,110 @@ impl StateMachine {
         Ok(AppliedState::DatabaseMeta(Change::new(None, None)))
     }
 
+    fn apply_rename_database_cmd(
```

Comment:
Should we reuse `apply_drop_database_cmd` and `apply_create_database_cmd` code? like `apply_rename_table_cmd`.

Reply:
> Should we reuse `apply_drop_database_cmd` and `apply_create_database_cmd` code? Like `apply_rename_table_cmd`.

I think renamed database should use old database's db_id. 


```SQL
create database old;
alter database old rename to new 
```
not equal with:

```SQL
create database old;
drop database old;
create database new;
```

Reply:
> I think renamed database should use old database's db_id.

I'm not sure, it seems that the id and name are bound now.

cc @drmingdrmer @ariesdevil

Reply:
Yep, actually in apply_rename_database_cmd

https://github.com/datafuselabs/databend/blob/a21fb5bb7da6579c0a80a5d2ead3d8f073b33e77/common/meta/raft-store/src/state_machine/sm.rs#L622
```rust
self.txn_create_table(txn_tree, db_id, table_id, &req.new_table_name, table_meta)?;
```

table_id is table_name's id and table_meta also from old table( &prev.as_ref().unwrap().data).

Reply:
> > I think renamed database should use old database's db_id.
> 
> I'm not sure, it seems that the id and name are bound now.
> 
> cc @drmingdrmer @ariesdevil

The db_id and table_id need to remain the same when changing the db/table meta

Reply:
> The db_id and table_id need to remain the same when changing the db/table meta

Thanks, I found it https://github.com/datafuselabs/databend/pull/4838

Reply:
@TCeason It looks like there is no conflict between reusing the code and not changing the db_id, It can also be refactored in another PR.

Reply:
> @TCeason It looks like there is no conflict between reusing the code and not changing the db_id, It can also be refactored in another PR.
Agree with you.